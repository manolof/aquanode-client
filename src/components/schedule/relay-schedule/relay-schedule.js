import PropTypes from 'prop-types';
import React, { Component } from 'react';

class RelaySchedule extends Component {
	type = 'relay';

	render() {
		const { schedule, resetSchedule, setSchedule } = this.props;

		return (
			<section className="schedule schedule__relay">
				<h4>Relay</h4>

				<article className="schedule__values">
					{schedule}
				</article>

				<footer className="schedule__footer">
					<button className="schedule__reset" onClick={() => resetSchedule(this.type)}>RESET</button>
					<button className="schedule__set" onClick={() => setSchedule(this.type, 'on')}>SET ON</button>
					<button className="schedule__set" onClick={() => setSchedule(this.type, 'off')}>SET OFF</button>
				</footer>
			</section>
		);
	}
}

RelaySchedule.propTypes = {
	schedule: PropTypes.node.isRequired,
	resetSchedule: PropTypes.func.isRequired,
	setSchedule: PropTypes.func.isRequired,
};

export {
	RelaySchedule,
};
