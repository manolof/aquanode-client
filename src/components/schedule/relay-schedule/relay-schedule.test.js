import { shallow } from 'enzyme';
import React from 'react';

import { RelaySchedule } from './relay-schedule';

describe('Relay Schedule Component', () => {
	const resetScheduleSpy = jest.fn();
	const setScheduleSpy = jest.fn();
	const mockSchedule = <p>Mock Schedule</p>;

	afterEach(() => {
		jest.resetAllMocks();
	});

	afterAll(() => {
		jest.restoreAllMocks();
	});

	it('should render', () => {
		const wrapper = shallow(
			<RelaySchedule schedule={mockSchedule} resetSchedule={resetScheduleSpy} setSchedule={setScheduleSpy}/>,
		);

		expect(wrapper.getElements()).toMatchSnapshot();
	});

	it(`should have functioning buttons that modify the relay' schedule`, () => {
		const wrapper = shallow(
			<RelaySchedule schedule={mockSchedule} resetSchedule={resetScheduleSpy} setSchedule={setScheduleSpy}/>,
		);

		expect(resetScheduleSpy).not.toHaveBeenCalled();
		wrapper.find('.schedule__reset').simulate('click');
		expect(resetScheduleSpy).toHaveBeenCalledTimes(1);

		expect(setScheduleSpy).not.toHaveBeenCalled();
		wrapper.find('.schedule__set').at(0).simulate('click');
		expect(setScheduleSpy).toHaveBeenCalledTimes(1);
		expect(setScheduleSpy).toHaveBeenLastCalledWith('relay', 'on');

		wrapper.find('.schedule__set').at(1).simulate('click');
		expect(setScheduleSpy).toHaveBeenCalledTimes(2);
		expect(setScheduleSpy).toHaveBeenLastCalledWith('relay', 'off');
	});
});
