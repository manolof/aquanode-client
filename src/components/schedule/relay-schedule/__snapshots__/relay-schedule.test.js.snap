// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Relay Schedule Component should render 1`] = `
Array [
  <section
    className="schedule schedule__relay"
  >
    <h4>
      Relay
    </h4>
    <article
      className="schedule__values"
    >
      <p>
        Mock Schedule
      </p>
    </article>
    <footer
      className="schedule__footer"
    >
      <button
        className="schedule__reset"
        onClick={[Function]}
      >
        RESET
      </button>
      <button
        className="schedule__set"
        onClick={[Function]}
      >
        SET ON
      </button>
      <button
        className="schedule__set"
        onClick={[Function]}
      >
        SET OFF
      </button>
    </footer>
  </section>,
]
`;
